netsim/modules/lag.py

```diff
@@ -50,186 +63,207 @@ def check_lag_config(node: str, linkname: str, topology: Box) -> bool:
   return True
 
 """
-check_same_pair - Verifies that the given member connects the same pair of nodes as the first
+check_mlag_support - check if the given node supports mlag
 """
-def check_same_pair(first_pair: list[str], member: Box) -> bool:
-  others = { n.node for n in member.interfaces if n.node not in first_pair }
-  if others:
-    log.error(f'All member links must connect the same pair of nodes({first_pair}), found {others}',
+def check_mlag_support(node: str, linkname: str, topology: Box) -> bool:
+  _n = topology.nodes[node]
+  features = devices.get_device_features(_n,topology.defaults)
+  if not features.lag.get('mlag',False):
+    log.error(f'Node {_n.name} ({_n.device}) does not support MLAG, cannot be part of peerlink or M-side of LAG {linkname}',
       category=log.IncorrectValue,
       module='lag')
     return False
   return True
 
+"""
+normalized_members - builds a normalized list of lag member links, checking various conditions
+"""
+def normalized_members(l: Box, topology: Box, peerlink: bool = False) -> list:
+  members = []                                      # Build normalized list of members
+  _name = "peerlink" if peerlink else "lag"
+  for idx,member in enumerate(l.lag.members):
+    member = links.adjust_link_object(member,f'{l._linkname}.{_name}[{idx+1}]',topology.nodes)
+    if 'lag' in member:                             # Catch potential sources for inconsistency
+      if peerlink or ('ifindex' not in member.lag): # ...but allow for custom bond numbering
```

Comment:
Maybe I got it all wrong, but this seems to accepts any LAG attribute on a member link as long as there's 'ifindex' on it. Also, why would you allow a user to specify bond interface index on the LAG member link?

Reply:
The ```ifindex``` may be needed in case of 2:2 mlag where there is a different bond index on each pair (we have some examples like that in our network).

Rewritten to only allow 'ifindex', no other lag attributes